src/UICrafter.Core/Components/Render/ComponentListRender.razor

```diff
@@ -0,0 +1,51 @@
+
+@if (componentList != null)
+{
+	@foreach (var component in componentList.UIComponents)
```

Comment:
This seems like you are doing the same thing, you should do only on foreach and in that

```cs
(var component in (componentList != null ? componentList.UIComponents : UIComponents ))
```

You can also extend this with more checking but do it within one scope to avoid copy paste